spec/apis/path-to-active-when.spec.js

```diff
@@ -2,182 +2,121 @@ import { pathToActiveWhen } from "single-spa";
+      "http://app.com/pathname/123/?query-string=1": true,
+      "http://app.com/pathname/123/extra": true,
+      "http://app.com/pathname/123/extra?query-string=1": true,
+      "http://app.com/pathname/123": false,
+      "http://app.com/pathname/123?query-string=1": false,
+    });
+
+    expectPathToMatch("/#/pathname", {
+      "http://app.com/#/pathname": true,
+      "http://app.com/#/pathname?query-string=1": true,
+      "http://app.com/#/pathname/": true,
+      "http://app.com/#/pathname/?query-string=1": true,
+      "http://app.com/#/pathname/anything/everything": true,
+      "http://app.com/#/pathname/anything/everything?query-string=1": true,
+      "http://app.com?query-string=1/#/pathname/anything/everything?query-string=1": true,
+      "http://app.com?query-string=1#/pathname/anything/everything?query-string=1": true,
+    });
+
+    expectPathToMatch("/#/pathname/:dynamic/notDynamic", {
+      "http://app.com/#/pathname/1/notDynamic": true,
+      "http://app.com/#/pathname/1/notDynamic/": true,
+      "http://app.com/#/pathname/1/notDynamicExtra": false,
+      "http://app.com/#/pathname//notDynamic/anything/everything": false,
+    });
+
+    expectPathToMatch("/pathname/:dynamic/notDynamic", {
+      "http://app.com/pathname/1/notDynamic": true,
+      "http://app.com/pathname/1/notDynamic/anything/everything": true,
+      "http://app.com/pathname//notDynamic": false,
+    });
+
+    expectPathToMatch("", {
+      "http://app.com": true,
+      "http://app.com/": true,
+      "http://app.com/anything/everything": true,
+    });
+
+    expectPathToMatch("/", {
+      "http://app.com/": true,
+      "http://app.com/anything/everything": true,
+    });
+
+    expectPathToMatch("/:dynamic/:dynamic", {
+      "http://app.com/1/1": true,
+      "http://app.com/1/1/": true,
+      "http://app.com/1//": false,
+      "http://app.com/1/": false,
+      "http://app.com/1": false,
+    });
+
+    expectPathToMatch("/:dynamic/:dynamic/anything/everything", {
+      "http://app.com/1/1/anything/everything": true,
+    });
+
+    expectPathToMatch("/pathname#/subpath/:dynamic", {
+      "http://app.com/pathname#/subpath/1/with/other/things": true,
+      "http://app.com/#/subpath/1/with/other/things": false,
+    });
+
+    expectPathToMatch("/#/subpath/:dynamic", {
+      "http://app.com/#/subpath/1/with/other/things": true,
+      "http://app.com/#/subpath/1": true,
+      "http://app.com/subpath/1/with/other/things": false,
+    });
+
+    // Impossible paths due to URL always goin from root 'app.com' -> 'app.com/'
+    expectPathToMatch("#/subpath/:dynamic", {
+      "http://app.com#/subpath/1/with/other/things": false,
+      "http://app.com#/subpath/1": false,
+    });
+
+    expectPathToMatch("pathname", {
+      "http://app.com/pathname": false,
+      "http://app.com/pathname/bleus": false,
+    });
+
+    expectPathToMatch(":dynamic/:dynamic", {
+      "http://app.com/1/1": false,
```

Comment:
Why is this false? In the [previous tests](https://github.com/single-spa/single-spa/pull/594/files#diff-a769be383da96d054df67ad869d0f04dL134), it was true

Reply:
The previous test was testing a different part of the code, just the regex generator... If `:dynamic/:dynamic` started with a `/`, this test would be true

Reply:
I see